### PR TITLE
Feature/en 4289 jayasimha

### PR DESCRIPTION
**Pull request by jayasimha-raghavan-unskript**
## Description
[EN-4289]
[EN-4315]
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published. 

### Documentation
Make sure that you have documented corresponding changes in this repository. 

[EN-4289]: https://unskript.atlassian.net/browse/EN-4289?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[EN-4315]: https://unskript.atlassian.net/browse/EN-4315?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Kubernetes/legos/k8s_get_cluster_health/k8s_get_cluster_health.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    for node in nodes.items():
+        # Lets check Node Pressure, more than 80%, will need to to raise an exception
+        cpu_usage = normalize_cpu(node.status.allocatable['cpu'])
+        cpu_capacity = normalize_cpu(node.status.capacity['cpu'])
+        mem_usage = normalize_memory(node.status.allocatable['memory'])
+        mem_capacity = normalize_memory(node.status.capacity['memory'])
+        cpu_usage_percent = (cpu_usage / cpu_capacity) * 100
+        mem_usage_percent = (mem_usage / mem_capacity) * 100 
+        if cpu_usage_percent >= 80 or mem_usage_percent >= 80:
+            raise Exception(f"Node {node.metadata.name} Experiencing High CPU {round(cpu_usage_percent,2)}% / MEM {round(mem_usage_percent,2)}% usage")
```
Comment (amit-chandak-unskript) [inline Kubernetes/legos/k8s_get_cluster_health/k8s_get_cluster_health.py:94]: thats not an exception, right? we are trying to get the list of pods/nodes which are not healthy, by raising exception, we are loosing the failed objects data.
Think about this check will be consumed by the remediation runbook, if we had one.

File: Kubernetes/legos/k8s_get_cluster_health/k8s_get_cluster_health.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        # Check the status of the Kubernetes pods
+        pods = pods_api.list_pod_for_all_namespaces()
+        pod_condition = ''
+        for pod in pods.items:
+            conditions = pod.status.conditions
+            for condition in conditions:
+                if condition.type == 'Ready' and condition.status == 'True':
+                    break
+            else:
+                pod_condition += f"\tPod {pod.metadata.namespace}/{pod.metadata.name} is not ready"
```
Comment (amit-chandak-unskript) [inline Kubernetes/legos/k8s_get_cluster_health/k8s_get_cluster_health.py:122]: Such kind of check is not at all consumable by the remediation runbook. Please think about how this check output will be used.

**Review by amit-chandak-unskript — CHANGES_REQUESTED** [review]
Requires work

File: Kubernetes/legos/k8s_get_cluster_health/k8s_get_cluster_health.py (hunk 1 above)
Comment (jayasimha-raghavan-unskript) [reply Kubernetes/legos/k8s_get_cluster_health/k8s_get_cluster_health.py]: Done. As we discussed returns dict with key `events`, `node_condition` and `unhealthy_pods` so downstream can consume the data.

File: Kubernetes/legos/k8s_get_cluster_health/k8s_get_cluster_health.py (hunk 2 above)
Comment (jayasimha-raghavan-unskript) [reply Kubernetes/legos/k8s_get_cluster_health/k8s_get_cluster_health.py]: Modified to return Python dict with the key `unhealthy_pod` that has a list of dictionaries having details of the unhealthy pods.

**Review by amit-chandak-unskript — APPROVED** [review]

**Pull request merged**